src/algorithms/base_algorithm.h

```diff
@@ -62,15 +62,15 @@ class BaseAlgorithm {
   //! Allocation for each datum, i.e. label of the cluster it belongs to
   std::vector<unsigned int> allocations;
   //! Hierarchy of the unique values that identify each cluster
-  std::vector<std::shared_ptr<BaseHierarchy>> unique_values;
+  std::vector<std::shared_ptr<AbstractHierarchy>> unique_values;
   //!
   Eigen::MatrixXd hier_covariates;
   //! Mixing object
   std::shared_ptr<BaseMixing> mixing;
   //!
   Eigen::MatrixXd mix_covariates;
   //!
-  bool update_hierarchy_params = false;
+  virtual bool update_hierarchy_params() { return false; }
```

Comment:
I realized that it wasn't working, try to run Neal3 and print 

```
std::cout << "update_hierarchy_params: " << update_hierarchy_params
            << std::endl;
```
in Neal2::sample_allocations; it would print '0'.

Reply:
I see. Using inheritance is certainly safer and brings no unexpected surprises anyway.